core/src/main/scala/algebra/Eq.scala

```diff
@@ -70,6 +70,19 @@ object Eq extends EqFunctions {
     }
 
   /**
+   * Create an `Eq` instance that defines equality via the conjunction
+   * of each of the given functions `fs`.
+   */
+  def fromAll[@sp A](fs : ((A,A) => Boolean)*) : Eq[A] =
+    new Eq[A] {
+      def eqv(x: A, y: A) =
+        fs.foldLeft(true) {
```

Comment:
I believe this behaves as documented. If any of the method returns false, the fold stops with false as a result. The only way to get true would be for all methods to return true. I'm not sure why you think this only looks for the first true.
I wrote another test to verify this behavior, but I'm not sure if I should add that, since it is a standard exemplary test, not a law test.